peer/tworandomchoices/tworandomchoices.go

```diff
@@ -103,24 +104,28 @@ func (l *twoRandomChoicesList) Choose(_ *transport.Request) peer.StatusPeer {
 	if j >= numSubs {
 		j -= numSubs
 	}
-	if l.pending(i) > l.pending(j) {
+	if l.subscribers[i].pending > l.subscribers[j].pending {
 		i = j
 	}
 	return l.subscribers[i].peer
 }
 
-func (l *twoRandomChoicesList) pending(index int) int {
-	return l.subscribers[index].pending
+func (l *twoRandomChoicesList) updatePendingRequestCountForPeer(s *subscriber, pendingRequestCount int) {
+	l.m.Lock()
+	defer l.m.Unlock()
+
+	s.pending = pendingRequestCount
```

Comment:
Because this RWMutex is on a list, shared by all its `[]subscriber`. This means a single subscriber update will block all others' update.
In terms of correctness, I think it is fine. During Choose(read), caller might just get a peer not with the least pending peer. 
The question here is on lock contention. According to [doc](https://golang.org/pkg/sync/#RWMutex):
```
Lock: If the lock is already locked for reading or writing, Lock blocks until the lock is available.
RLock: a blocked Lock call excludes new readers from acquiring the lock
```
If there is lots of updates happening (per request finish), will we cause a blocking issue on Choose here?

Reply:
You are totally right on this! Behavior speaking this is good. However performance speaking we can improve this by using atomic instead of mutex.